### PR TITLE
Ensures new deployments are started

### PR DESCRIPTION
**Pull request by johnsimons**
Fixes https://github.com/OctopusDeploy/Issues/issues/6572

### CODE REVIEW

File: source/Calamari.Tests/DeployAzureCloudServiceCommandFixture.cs [hunk 1]
```diff
@@ -84,6 +89,12 @@ public async Task Deploy_Package_To_Stage()
                 await client.HostedServices.DeleteAsync(serviceName);
             }
 
+            async Task EnsureDeploymentIsRunning(DeploymentStatus requiredStatus)
```
Comment (flin-8) [inline source/Calamari.Tests/DeployAzureCloudServiceCommandFixture.cs:92]: Rename to `AssertRunningStatus`

File: source/Calamari.Tests/DeployAzureCloudServiceCommandFixture.cs [hunk 2]
```diff
@@ -67,9 +66,15 @@ public async Task Deploy_Package_To_Stage()
                 await client.HostedServices.CreateAsync(new HostedServiceCreateParameters(serviceName, "test") { Location = "West US" });
 
                 await Deploy();
+                await EnsureDeploymentIsRunning(DeploymentStatus.Running);
 
-                //Run again to test upgrading an existing slot
+                // Suspend state
+                await client.Deployments.UpdateStatusByDeploymentSlotAsync(serviceName, deploymentSlot, new DeploymentUpdateStatusParameters(UpdatedDeploymentStatus.Suspended));
+
+                //Run again to test upgrading an existing slot nad status should not change
```
Comment (flin-8) [inline source/Calamari.Tests/DeployAzureCloudServiceCommandFixture.cs:74]: typo `nad`

Comment (johnsimons) [reply source/Calamari.Tests/DeployAzureCloudServiceCommandFixture.cs]: Fixed

File: source/Calamari.Tests/DeployAzureCloudServiceCommandFixture.cs (hunk 1 above)
Comment (johnsimons) [reply source/Calamari.Tests/DeployAzureCloudServiceCommandFixture.cs]: Fixed

File: source/Calamari.Tests/DeployAzureCloudServiceCommandFixture.cs [hunk 3]
```diff
@@ -145,7 +156,7 @@ public async Task Deploy_Package_To_Stage_And_Swap()
                                                          context.Variables.Add(SpecialVariables.Action.Azure.UseCurrentInstanceCount, bool.FalseString);
                                                          context.Variables.Add(SpecialVariables.Action.Azure.DeploymentLabel, "v1.0.0");
 
-                                                         context.WithPackage(pathToPackage, "Octopus.Sample.AzureCloudService", "5.8.2");
+                                                         context.WithPackage(pathToPackage, "Octopus.Sample.AzureCloudService", "6.0.0");
                                                      })
                                         .Execute();
```
Comment (flin-8) [inline source/Calamari.Tests/DeployAzureCloudServiceCommandFixture.cs:162]: Should assert prod exists too, is it worth asserting the status here?

File: source/Calamari/DeployAzureCloudServicePackageBehaviour.cs [hunk 4]
```diff
@@ -113,7 +113,8 @@ static async Task<bool> Exists(IComputeManagementClient azureClient, string serv
                                                               Label = deploymentLabel,
                                                               Configuration = File.ReadAllText(configurationFile),
                                                               PackageUri = new Uri(packageUri),
-                                                              Name = Guid.NewGuid().ToString("N")
+                                                              Name = Guid.NewGuid().ToString("N"),
+                                                              StartDeployment = true
```
Comment (flin-8) [inline source/Calamari/DeployAzureCloudServicePackageBehaviour.cs:117]: 👍

File: source/Calamari.Tests/DeployAzureCloudServiceCommandFixture.cs (hunk 3 above)
Comment (johnsimons) [reply source/Calamari.Tests/DeployAzureCloudServiceCommandFixture.cs:158]: Done

**Review by flin-8 — APPROVED** [review]
Looks good, much better test coverage now

**Pull request merged**